src/swap/useSwapQuote.ts

```diff
@@ -67,42 +74,132 @@ function createBaseApproveTransaction(
       ? BigInt(new BigNumber(buyAmount).times(guaranteedPrice).toFixed(0, 0))
       : BigInt(sellAmount)
 
-  const data = encodeFunctionData({
-    abi: erc20.abi,
-    functionName: 'approve',
-    args: [allowanceTarget as Address, amountToApprove],
-  })
-
-  const approveTx: TransactionRequestCIP42 & { amountToApprove: bigint } = {
-    from: from as Address,
-    to: sellTokenAddress as Address,
-    data,
-    type: 'cip42',
-    amountToApprove,
+  // Approve transaction if the sell token is ERC-20
+  if (allowanceTarget !== zeroAddress && fromToken.address) {
```

Comment:
do we want to add a check for how much is already approved, so we don't need to do this and save the user some gas cost if they've previously done a bigger swap using the same allowanceTarget?

Reply:
That's a great idea! Keeping a note for a follow up PR.